onvif/client.py

```diff
@@ -289,6 +293,37 @@ async def close(self):
         for service in self.services.values():
             await service.close()
 
+    def get_snapshot(self, url, basic_auth):
+        """Get a snapshot image from the camera."""
+        if url is None:
+            return None
+
+        auth = None
+        if self.user and self.passwd:
+            if basic_auth:
+                auth = HTTPBasicAuth(self.user, self.passwd)
+            else:
+                auth = HTTPDigestAuth(self.user, self.passwd)
+
+        try:
+            response = requests.get(url, timeout=5, auth=auth)
```

Comment:
Since everything else in this library is async, can we wrap this in an executor and make it async as well? See https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.loop.run_in_executor

Reply:
ok, I'll probably increase the minimum python version to 3.7 if doing this, though.